core/src/test/java/org/elasticsearch/common/settings/loader/JsonSettingsLoaderTests.java

```diff
@@ -50,4 +59,16 @@ public void testSimpleJsonSettings() throws Exception {
         assertThat(settings.getAsArray("test1.test3")[0], equalTo("test3-1"));
         assertThat(settings.getAsArray("test1.test3")[1], equalTo("test3-2"));
     }
+
+    @Test
+    public void testDuplicateKeysThrowsException() {
+        expectedException.expect(SettingsException.class);
+        expectedException.expectCause(Matchers.<Throwable>instanceOf(ElasticsearchParseException.class));
```

Comment:
or junit for that matter. try/catch is much more readable (and the way most other tests do this)